### DISCUSSION TITLE
Multichannel ADC DMA conversion sample for STM32F7 evice

### DISCUSSION THREAD

**Discussion by vailatis** [stm32duino/Arduino_Core_STM32#1650] [Libraries]
I've looked inside any part of the code, but examples are only available for F1 and F4 processors that uses maple library. For an F7 processor, the only examples about ADC and a simple "analogRead" that give very slow performance, i cannot read faster that 2 mSec intervals.
Are there any example, or something to follow in order to make an STM32F746ZG Nucleo-144 board to be able to create a Multichannel ADC with DMA code in order to sample 2 or more ADC channels in a faster way?

**Comment by fpistm** [comment]
Hi @vailatis 
You're right, currently this core does not provide DMA support while the Maple core use it. 
Anyway, you can use HAL at sketch level to perform what you need.
As example you can check this topic:
https://www.stm32duino.com/viewtopic.php?f=41&t=110

And try to do the same using an example F7 project like this one: https://github.com/STMicroelectronics/STM32CubeF7/tree/master/Projects/STM32F746ZG-Nucleo/Examples/ADC/ADC_RegularConversion_DMA
